app/controllers/time_slot_controller.rb

```diff
@@ -15,7 +15,7 @@ def schedule
     return render 'patients/not_allowed' unless @patient.can_schedule?
 
     Appointment.transaction do
-      if Appointment.where(start: start_time, ubs: @ubs).where.not(patient_id: nil).count == @ubs.appointments_per_time_slot
+      if Appointment.where(start: start_time, ubs: @ubs, patient_id: nil).count.zero?
```

Comment:
isso é porque hj temos a situação que o numer de slots por horario varia e o numero de slots salvo na ubs esta obsoleto (pois é fixo), ocorreria que se na ubs fosse == 1 e ja tivesse uma das vagas do time slot preencjhida aqui retornava que n tinha mais horario disponivel

Reply:
Boa!

Um nitpick apenas:

```suggestion
      unless Appointment.where(start: start_time, ubs: @ubs, patient_id: nil).exists?
```

Counts são razoavelmente custosos :grimacing: